src/Workspaces/Core/Portable/Workspace/Solution/ProjectState_Checksum.cs

```diff
@@ -43,8 +43,9 @@ public Task<ProjectStateChecksums> GetStateChecksumsAsync(CancellationToken canc
                 var infoChecksum = serializer.CreateChecksum(ProjectInfo.Attributes, cancellationToken);
 
                 // these compiler objects doesn't have good place to cache checksum. but rarely ever get changed.
-                var compilationOptionsChecksum = SupportsCompilation ? ChecksumCache.GetOrCreate(CompilationOptions, _ => serializer.CreateChecksum(CompilationOptions, cancellationToken)) : Checksum.Null;
-                var parseOptionsChecksum = SupportsCompilation ? ChecksumCache.GetOrCreate(ParseOptions, _ => serializer.CreateChecksum(ParseOptions, cancellationToken)) : Checksum.Null;
+                // in some unit tests or project which doesn't support Roslyn semantic model, options can be null
+                var compilationOptionsChecksum = CompilationOptions != null ? ChecksumCache.GetOrCreate(CompilationOptions, _ => serializer.CreateChecksum(CompilationOptions, cancellationToken)) : Checksum.Null;
+                var parseOptionsChecksum = ParseOptions != null ? ChecksumCache.GetOrCreate(ParseOptions, _ => serializer.CreateChecksum(ParseOptions, cancellationToken)) : Checksum.Null;
```

Comment:
in unit test, ParseOptions can be null since people create ProjectInfo without giving in options.

http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/AdhocWorkspace.cs,66